File: Programmers/SeungDo/2주차(Brute_force(완전 탐색)/최소직사각형.cpp
#include <string>
#include <vector>

// 가로 길이가 가장 큰 명함을 찾은뒤 세로길이가 가장 큰 명함을 찾고 answer에 곱해줌

using namespace std;
void tempFunc(vector<vector<int>> &sizes , int i) { // 가로길이에 가장 큰값으로 몰아줌
    int temp = 0;
    if (sizes[i][1] > sizes[i][0]) {
        temp = sizes[i][1];
        sizes[i][1] = sizes[i][0];
        sizes[i][0] = temp;
    }
}
int solution(vector<vector<int>> sizes) {
    int answer = 0, max1 = 0, max2 = 0; // max: 큰값  
    for (int i = 0; i < sizes.size(); i++) {     // 가장 큰 가로 길이 찾음
        tempFunc(sizes, i);       // 만약 가로길이나 세로길이가 반대쪽 길이보다 크면 큰수로 바꿔줌 
        if (max1 < sizes[i][0]) max1 = sizes[i][0];    // 가장 가로 큰값 찾음  
        if (max2 < sizes[i][1]) max2 = sizes[i][1];    // 가장 세로 큰값 찾음
    }
    answer = max1 * max2;     // 나온 max값을 answer에 더해줌

    return answer;
}

File: Programmers/JeongHyun/2주차 - Brute_force(완전 탐색)/최소직사각형.js
function solution(sizes) {
    let answer = 0;
    let n =0;
    let rotated = sizes.map(([w,h])=>w<h?[h,w]:[w,h]);
    //rotated 는 sizes의 배열 내에서 w와 h 중 더 큰 수를 왼쪽으로 옮기기 위한 변수입니다.
    rotated.sort((a,b)=>{return b[0]-a[0]});
    //sort를 통해 내림차순으로 정렬
    let max = -1;
    // [1,2] sizes의 1의 원소 중 큰값을 찾기 위한 max값 임의 설정
    for(let i=0;i<rotated.length;++i){
        // rotated의 행(↓)만을 탐색하며 큰수 비교 
        if(rotated[i][1]>max)max = rotated[i][1];
        n++;
    }
    //rotated[0][0]이 가장 큰 원소
    answer = rotated[0][0]*max;
    return answer;
}
